modules/util/net.go

```diff
@@ -0,0 +1,19 @@
+// Copyright 2021 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package util
+
+import (
+	"net"
+)
+
+// IsIPPrivate for net.IP.IsPrivate. TODO: replace with `ip.IsPrivate()` if min go version is bumped to 1.17
+func IsIPPrivate(ip net.IP) bool {
```

Comment:
Why not just move this directly to `modules/hostmatcher/`?

Reply:
Other packages also need it:

https://github.com/go-gitea/gitea/pull/17510/files#diff-29b9acf1837f54ae83957698e86612ff65c2a18b859b0d442f06c1907a21b09dR92

And it is really a util function, which is to be replace with IP.IsPrivate in Go1.7.